hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/operations/PartitionIteratingOperation.java

```diff
@@ -180,15 +180,10 @@ public Operation create(int partitionId) {
         private void executeOperations(PartitionAwareOperationFactory givenFactory) {
             final NodeEngine nodeEngine = getNodeEngine();
 
-            final PartitionAwareOperationFactory factory = givenFactory.createFactoryOnRunner(nodeEngine);
-
-            int[] operationFactoryPartitions = factory.getPartitions();
-            partitions = operationFactoryPartitions == null ? partitions : operationFactoryPartitions;
```

Comment:
I have checked that the returned partition override was effective for `(HD)PartitionWideEntryWithPredicateOperationFactory` only, but it would be better if a second pair of eyes double check that ;)